archiver/biosamples.py

```diff
@@ -4,6 +4,19 @@
 from conversion.json_mapper import JsonMapper
 from conversion.post_process import format_date, default_to
 
+ontology_url = 'http://purl.obolibrary.org/obo/{}'
```

Comment:
is this always the ontology url?

Reply:
i think you need to use `expand_curie` method 🤔

Reply:
i do remember we have issues about ontology validation before 🤔  i wonder if this is related to that.

Reply:
and we don't seem to encounter that now since we don't have that happening in the conversion now.

Reply:
it's not always like that, you need to retrieve the iri https://www.ebi.ac.uk/ols/docs/api

Reply:
And also as @aaclan-ebi said we used to have problems, because for bioSamples it needs to be a "defining ontology"

Reply:
implanted a map from `obo_id` to `iri` using ontology api, fallback to the above url if no `iri` is found.

Reply:
the expand curie method is strictly the wrong thing to do
From a sample size of 1,184 organs and organ parts:
 - lookups to expand the curie on the text of our ontology give a different iri 1,117 times
 - they match the same url 70 times
 - lookups failed with no match found 74 times

Reply:
changed the fallback to be a standard text attribute of the ontology.text if the iri lookup fails